src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -352,120 +354,145 @@ public static string Combine(params string[] paths)
             return StringBuilderCache.GetStringAndRelease(finalPath);
         }
 
-        private static string CombineNoChecks(string path1, string path2)
+        /// <summary>
+        /// Combines two paths. Does no validation of paths, only concatenates the paths
+        /// and places a directory separator between them if needed.
+        /// </summary>
+        private static string CombineNoChecks(ReadOnlySpan<char> first, ReadOnlySpan<char> second)
         {
-            if (path2.Length == 0)
-                return path1;
+            if (first.Length == 0)
+                return second.Length == 0
+                    ? string.Empty
+                    : new string(second);
 
-            if (path1.Length == 0)
-                return path2;
+            if (second.Length == 0)
+                return new string(first);
```

Comment:
You should create a follow up issue to try and get rid of the `new string`'s that you introduced in these helpers. Better to deal with it separately as they should be corner cases.

Reply:
okay I will do that . should I open to change the implementation for the NormalizeDirectorypath to not use stringbuilder ?

Reply:
Having issues to track pending work is never a bad thing. :)